derp/derp_server.go

```diff
@@ -1683,6 +1683,10 @@ func (c *sclient) sendLoop(ctx context.Context) error {
 			werr = c.sendKeepAlive()
 			continue
 		default:
+			// If we happen to have burned too much time in the goroutine and the derper
+			// is really busy, there's a possibility that the write doesn't start until
+			// the flush, so extend the deadline.
+			c.setWriteDeadline()
```

Comment:
Seems pretty unlikely?

Reply:
wait, each of the write calls above already does this. so I can't see how this does anything

Reply:
Each of the write calls is a buffered write, extending the socket write deadline should only matter just before flush?

Reply:
i.e. we could remove all the extensions for the other calls, and just do it here